extensions/amp-story/1.0/test/test-animation.js

```diff
@@ -0,0 +1,650 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {match} from 'sinon'; // eslint-disable-line local/no-import
+
+import {
+  AnimationManager,
+  AnimationRunner,
+  AnimationSequence,
+} from '../animation';
+import {Deferred} from '../../../../src/utils/promise';
+import {Services} from '../../../../src/services';
+import {WebAnimationPlayState} from '../../../amp-animation/0.1/web-animation-types';
+import {htmlFor, htmlRefs} from '../../../../src/static-template';
+import {layoutRectLtwh} from '../../../../src/layout-rect';
+import {presets} from '../animation-presets';
+import {scopedQuerySelectorAll} from '../../../../src/dom';
+import {toArray} from '../../../../src/types';
+
+const querySelectorAllAnimateIn = (element) =>
+  toArray(scopedQuerySelectorAll(element, '[animate-in]'));
+
+describes.realWin('amp-story animations', {}, (env) => {
+  let html;
+  let ampdoc;
+
+  const nextTick = () =>
+    new Promise((resolve) => env.win.setTimeout(resolve, 0));
+
+  beforeEach(() => {
+    html = htmlFor(env.win.document);
+    ampdoc = {win: env.win};
```

Comment:
You can use a real `AmpDoc`
```
import {AmpDocSingle} from '../../../../src/service/ampdoc-impl';

ampdoc = new AmpDocSingle(env.win);
```

Reply:
We only need the window reference, do you see advantages to this?

Reply:
Not necessarily in this one case, it's just a good practice. I'm always afraid of faking things in unit tests as it doesn't protect you from changing APIs. If it takes you hours of debugging it's obviously not worth it, but if it comes for free 🤷‍♂️

Reply:
I mostly worry about the instantiation cost, but for an AmpDoc I don't feel super strongly. Changed.